src/db/schemas/settlement.model.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  pgEnum,
+  pgTable,
+  real,
+  timestamp,
+  varchar,
+} from "drizzle-orm/pg-core";
+import { createInsertSchema, createSelectSchema } from "drizzle-zod";
+import type { z } from "zod";
+
+import { splitTypeArr } from "@/common/enums";
+
+import groupModel from "./group.model";
+import userModel from "./user.model";
+
+export const splitTypeEnum = pgEnum("splitType", splitTypeArr);
+
+const settlementModel = pgTable("settlement", {
+  id: varchar({ length: 60 })
+    .$defaultFn(() => Bun.randomUUIDv7())
+    .primaryKey()
+    .notNull(),
+  payerId: varchar({ length: 60 })
+    .notNull()
+    .references(() => userModel.id, {
+      onDelete: "cascade",
+      onUpdate: "cascade",
+    }),
+  receiverId: varchar({ length: 60 })
+    .notNull()
+    .references(() => userModel.id, {
+      onDelete: "cascade",
+      onUpdate: "cascade",
+    }),
+  groupId: varchar({ length: 60 }).references(() => groupModel.id, {
+    onDelete: "cascade",
+    onUpdate: "cascade",
+  }),
+  amount: real().notNull(),
+  createdAt: timestamp().notNull().defaultNow(),
+  updatedAt: timestamp().notNull().defaultNow(),
+});
+
+// Schema for selecting/inserting a settlement
+export const selectSettlementSchema = createSelectSchema(settlementModel, {
+  id: schema => schema.id.describe("Unique identifier for the settlement"),
+  payerId: schema =>
+    schema.payerId.describe("Reference to the user who is payer"),
+  receiverId: schema =>
+    schema.payerId.describe("Reference to the user who ower "),
```

Comment:
Please check the spelling, a good VS Code extension is called Spell checker which can be installed from https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker

Reply:
Spelling is correct. I checked in google. The term `ower` means who `owes` the amount. There was some grammer mistake. Corrected!